### PR TITLE
Upgrade and reenable reek

### PR DESCRIPTION
**Pull request by backus**

### CODE REVIEW

File: tasks/metrics/reek.rake [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # encoding: utf-8
 
 namespace :metrics do
-  task :reek do
-    $stderr.puts 'Reek is disabled due defunct AST processing'
+  require 'reek/rake/task'
+
+  project = Devtools.project
```
Comment (mbj) [inline tasks/metrics/reek.rake]: We do not use this lvar and the one below. We can just kill that two lines.

**Comment by mbj** [comment]
@backus I left a small notice about unused lvars. Code wise I'm fine otherwise. We are discussing other integration issues in channel right now. Ready for your next pass.

**Comment by backus** [comment]
I'll take a look tonight

**Comment by backus** [comment]
Here is the result of running reek against rubyspec: https://gist.github.com/backus/d6b0849b5f509592142f

I ran reek against each file individually so 1) it would be clear which files caused error and 2) an error parsing one file wouldn't obfuscate an error from another.  

Two errors raised were:

1. invalid byte sequence in UTF-8 (EncodingError) (from [core/dir/pwd_spec.rb](https://github.com/rubyspec/rubyspec/blob/archive/core/dir/pwd_spec.rb))
2. invalid encoding symbol (EncodingError) (from [core/symbol/casecmp_spec.rb](https://github.com/rubyspec/rubyspec/blob/archive/core/symbol/casecmp_spec.rb)).

both parser issues

**Comment by backus** [comment]
Ready for another pass @mbj

**Comment by mbj** [comment]
> both parser issues

I do not agree with that statement. These are NOT issues by parser. These are deliberately part of the rubinius corpus to make sure examples of weird encodings are coverd.

**Comment by backus** [comment]
> I do not agree with that statement.

Fair enough.  What I mainly meant was these are issues not core to the reek codebase

**Comment by mbj** [comment]
> Fair enough. What I mainly meant was these are issues not core to the reek codebase

I'm fine with that conclusion.

**Pull request merged**